### PR TITLE
Add "enum" for countries

### PR DESCRIPTION
**Pull request by zachantre**
Fixed timezone in openwrt backend
Addition of countries and selection from menu

### CODE REVIEW

**Comment by nemesisdesign** [comment]
Hi @zachantre, thanks for your contribution.
There's one thing I don't understand, what exactly do you want to fix in the timezone setting?

**Comment by zachantre** [comment]
Well, openwrt timezone need time format... right now it submits continent/city which is not proper for openwrt. With fix, on site you see continent/city but submit the actual timezone

**Comment by nemesisdesign** [comment]
That shouldn't be an issue because the library supplies the correct value upon rendering/generation.

But I would like to address the mismatch and so the implementation is cleaner, I'm trying your patch but lots of things go wrong.

Have you tried running unit tests as explained in the README?

Federico

**Comment by nemesisdesign** [comment]
I see many failing tests with an error which I think it's hard to overcome: OpenWRT has quite a few timezones of different areas which have the same value, but specifying an ``enum`` which has duplicate values is not allowed by jsonschema RFC.

Example:

```
    (
        "Africa/Gaborone",
        "CAT-2"
    ),
    (
        "Africa/Harare",
        "CAT-2"
    ),
```

Supplying a list which has duplicate values to ``enum`` breaks the schema validation.

To understand what I'm trying to say, run this command:

```
nosetests -x tests.openwrt.test_backend
```

`-x` means stop at the first failure (otherwise you would get a ton of errors and you would be overwhelmed.

The other change to countries should easier to include.

File: netjsonconfig/schema.py [hunk 1]
```diff
@@ -782,6 +783,9 @@
                 "country": {
                     "type": "string",
                     "maxLength": 2,
+                    "default": "IT",
```
Comment (nemesisdesign) [inline netjsonconfig/schema.py]: remove this line please, there's no reason for which the default country should be Italy ;-)

**Comment by zachantre** [comment]
I see... Sorry for my mistake, I have just started learning python.
I will fix it and recommit.
I like openwisp2 project very much, it's very innovative and have much space for a lot of features.

**Comment by nemesisdesign** [comment]
Great! We just need to cleanup a few style issues before I can merge your patch.

**Comment by nemesisdesign** [comment]
See the failing build:
https://travis-ci.org/openwisp/netjsonconfig/jobs/157587560

You just have to remove a few trailing spaces from the indicated lines.

Then when you are done, do the following to squash your commits:

```
git rebase -i HEAD~4
```

Leave the last commit untouched and set the previous ones as "squash", then provide a better commit message and save, then push to your repository using `git push --force` and we are done.

Thank you @zachantre!

**Comment by zachantre** [comment]
On my pc I had some issues with tests... not properly set virtual env... 
Anyway, after some changes, travis shows everything fine... :)
I also removed timezones enum as suggested since the library supplies the correct value upon rendering

**Comment by nemesisdesign** [comment]
Thank you @zachantre, I squashed the commits and merged manually in d23d8fb. 👍

**Pull request closed without merging**